File: ResQNet_Project/resqnet-backend/src/main/java/com/pu_deltaforce/resqnet_backend/model/BloodBank.java
package com.pu_deltaforce.resqnet_backend.model;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "blood_banks")
@Data
public class BloodBank {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String donorName;

    @Column(nullable = false)
    private String phone;

    @Column(nullable = false)
    private String area;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private BloodGroup bloodGroup;

    private boolean available;

    public enum BloodGroup {
        A_POSITIVE, A_NEGATIVE,
        B_POSITIVE, B_NEGATIVE,
        O_POSITIVE, O_NEGATIVE,
        AB_POSITIVE, AB_NEGATIVE
    }

    @PrePersist
    public void prePersist() {
        this.available = true;
    }
}

File: ResQNet_Project/resqnet-frontend/src/app/core/services/blood-bank.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../../environments/environment';

export interface BloodDonor {
  id?: number;
  donorName: string;
  bloodGroup: 'A_POSITIVE' | 'A_NEGATIVE' | 'B_POSITIVE' | 'B_NEGATIVE' |
              'O_POSITIVE' | 'O_NEGATIVE' | 'AB_POSITIVE' | 'AB_NEGATIVE';
  area: string;
  phone: string;
  available?: boolean;
}

@Injectable({ providedIn: 'root' })
export class BloodBankService {
  private apiUrl = `${environment.apiUrl}/blood-bank`;

  constructor(private http: HttpClient) {}

  getByBloodGroup(group: string) {
    return this.http.get<BloodDonor[]>(`${this.apiUrl}/${group}`);
  }

  getByGroupAndArea(group: string, area: string) {
    return this.http.get<BloodDonor[]>(`${this.apiUrl}/${group}/${area}`);
  }

  create(data: BloodDonor) {
    return this.http.post<BloodDonor>(this.apiUrl, data);
  }

  toggleAvailability(id: number) {
    return this.http.put(`${this.apiUrl}/${id}/toggle-availability`, {});
  }
}